core/services/gateway/handlers/functions/subscriptions.go

```diff
@@ -87,7 +98,8 @@ func (s *onchainSubscriptions) Start(ctx context.Context) error {
 			return errors.New("OnchainSubscriptionsConfig.UpdateRangeSize must be greater than 0")
 		}
 
-		s.closeWait.Add(1)
+		s.closeWait.Add(2)
+		go s.loadCachedSubscriptions()
```

Comment:
Both goroutines call s.subscriptions.UpdateSubscription() so we have unnecessary races. I think we should populate the cache from the DB before we start fetching from the chain.

Reply:
ok will keep it sequential before fetching from the contract 👍🏼